dali/operators/image/convolution/gaussian_blur.cc

```diff
@@ -0,0 +1,309 @@
+ * * scalar argument - it will be replicated `result.size()` times
+ *
+ * TODO(klecki): we may want to make this a generic utility and propagate the span-approach to
+ * the rest of the related argument gettters
+ */
+template <typename T>
+void GetGeneralizedArg(span<T> result, const std::string name, int sample_idx, const OpSpec& spec,
+                       const ArgumentWorkspace& ws) {
+  int argument_length = result.size();
+  if (spec.HasTensorArgument(name)) {
+    const auto& tv = ws.ArgumentInput(name);
+    const auto& tensor = tv[sample_idx];
+    DALI_ENFORCE(tensor.shape().sample_dim() == 1,
+                 make_string("Argument ", name, " for sample ", sample_idx,
+                             " is expected to be 1D, got: ", tensor.shape().sample_dim(), "."));
+    DALI_ENFORCE(tensor.shape()[0] == 1 || tensor.shape()[0] == argument_length,
+                 make_string("Argument ", name, " for sample ", sample_idx,
+                             " is expected to have shape equal to {1} or {", argument_length,
+                             "}, got: ", tensor.shape(), "."));
+    if (tensor.shape()[0] == 1) {
+      for (int i = 0; i < argument_length; i++) {
+        result[i] = tensor.data<T>()[0];
+      }
+    } else {
+      memcpy(result.data(), tensor.data<T>(), sizeof(T) * argument_length);
+    }
+    return;
+  }
+  std::vector<T> tmp;
+  // we already handled the argument input, this handles spec-related arguments only
+  GetSingleOrRepeatedArg(spec, tmp, name, argument_length);
+  memcpy(result.data(), tmp.data(), sizeof(T) * argument_length);
+}
+
+template <int axes>
+GaussianBlurParams<axes> GetSampleParams(int sample, const OpSpec& spec,
+                                         const ArgumentWorkspace& ws) {
+  GaussianBlurParams<axes> params;
+  GetGeneralizedArg<float>(make_span(params.sigmas), kSigmaArgName, sample, spec, ws);
+  GetGeneralizedArg<int>(make_span(params.window_sizes), kWindowSizeArgName, sample, spec, ws);
+  for (int i = 0; i < axes; i++) {
+    DALI_ENFORCE(
+        !(params.sigmas[i] == 0 && params.window_sizes[i] == 0),
+        make_string("`sigma` and `window_size` shouldn't be 0 at the same time for sample: ",
+                    sample, ", axis: ", i, "."));
+    DALI_ENFORCE(params.sigmas[i] >= 0,
+                 make_string("`sigma` must have non-negative values, got ", params.sigmas[i],
+                             " for sample: ", sample, ", axis: ", i, "."));
+    DALI_ENFORCE(params.window_sizes[i] >= 0,
+                 make_string("`window_size` must have non-negative values, got ",
+                             params.window_sizes[i], " for sample: ", sample, ", axis : ", i, "."));
+    if (params.window_sizes[i] == 0 && params.sigmas[i] > 0.f) {
+      params.window_sizes[i] = SigmaToDiameter(params.sigmas[i]);
+    } else if (params.sigmas[i] == 0.f && params.window_sizes[i] > 0) {
+      params.sigmas[i] = DiameterToSigma(params.window_sizes[i]);
+    }
```

Comment:
Missing else - both sigma and window size specified. It's an important feature.

Reply:
It's not missing. If the user provided both sigma and window, they are already in params.sigmas and params.window_sizes. If they are positive, we just leave them to be.

This if/else if covers a case when one of them is `0` - which means calculate the value based on the other one.

I think I can remove the checks for one of the params being positive, as it's already enforced that they are non-negative and can't both be 0.